tests/lit-tests/2280.ispc

```diff
@@ -0,0 +1,38 @@
+// RUN:     %{ispc} %s      --target=host --nostdlib -E                                                              | grep -v -e '^// ' | FileCheck %s -check-prefix=CHECK_PASS_0
+// RUN:     %{ispc} %s -o - --target=host --nostdlib    --emit-llvm-text                                                                 | FileCheck %s -check-prefix=CHECK_PASS_1
+// RUN: not %{ispc} %s      --target=host --nostdlib -E                                               -DDO_ELSE 2>&1 | grep -v -e '^// ' | FileCheck %s -check-prefix=CHECK_ERROR
+// RUN: not %{ispc} %s -o - --target=host --nostdlib    --emit-llvm-text                              -DDO_ELSE 2>&1                     | FileCheck %s -check-prefix=CHECK_ERROR
+
+// RUN:     %{ispc} %s      --target=host --nostdlib -E                  --ignore-preprocessor-errors                | grep -v -e '^// ' | FileCheck %s -check-prefix=CHECK_PASS_0
+// RUN:     %{ispc} %s -o - --target=host --nostdlib    --emit-llvm-text --ignore-preprocessor-errors                                    | FileCheck %s -check-prefix=CHECK_PASS_1
+// RUN:     %{ispc} %s      --target=host --nostdlib -E                  --ignore-preprocessor-errors -DDO_ELSE      | grep -v -e '^// ' | FileCheck %s -check-prefix=CHECK_PASS_2
+// RUN:     %{ispc} %s -o - --target=host --nostdlib    --emit-llvm-text --ignore-preprocessor-errors -DDO_ELSE                          | FileCheck %s -check-prefix=CHECK_PASS_3
+
+
```

Comment:
Additional testing:
> warning when the suffix of the output file is not *.i or *.ispi
> 
> behavior with multi-target. This will need to be a separate test, as it will be x86 specific (as you will need to specify specific targets). Here's the list of pre-defined macros: https://ispc.github.io/ispc.html#the-preprocessor you can use one of them (for example TARGET_WIDTH) in combination with targets of different width like --target=sse4-i32x4,avx2-i32x8 and check how the parsing was done in output files. Also as it's not just a bug fix, but a feature, would be good to give a meaningful name to the test, like preprocessor.ispc / preprocessor_multi_target.ispc
@dbabokin